src/neo/SmartContract/Native/FungibleToken.cs

```diff
@@ -56,35 +56,36 @@ protected FungibleToken()
             Manifest.Abi.Events = events.ToArray();
         }
 
-        internal protected virtual void Mint(ApplicationEngine engine, UInt160 account, BigInteger amount, bool callOnPayment)
+        internal async ContractTask Mint(ApplicationEngine engine, UInt160 account, BigInteger amount, bool callOnPayment)
         {
             if (amount.Sign < 0) throw new ArgumentOutOfRangeException(nameof(amount));
             if (amount.IsZero) return;
             StorageItem storage = engine.Snapshot.GetAndChange(CreateStorageKey(Prefix_Account).Add(account), () => new StorageItem(new TState()));
             TState state = storage.GetInteroperable<TState>();
-            OnBalanceChanging(engine, account, state, amount);
+            await OnBalanceChanging(engine, account, state, amount);
             state.Balance += amount;
             storage = engine.Snapshot.GetAndChange(CreateStorageKey(Prefix_TotalSupply), () => new StorageItem(BigInteger.Zero));
             storage.Add(amount);
-            PostTransfer(engine, null, account, amount, StackItem.Null, callOnPayment);
+            await PostTransfer(engine, null, account, amount, StackItem.Null, callOnPayment);
         }
 
-        internal protected virtual void Burn(ApplicationEngine engine, UInt160 account, BigInteger amount)
+        internal protected void Burn(ApplicationEngine engine, UInt160 account, BigInteger amount)
         {
             if (amount.Sign < 0) throw new ArgumentOutOfRangeException(nameof(amount));
             if (amount.IsZero) return;
             StorageKey key = CreateStorageKey(Prefix_Account).Add(account);
             StorageItem storage = engine.Snapshot.GetAndChange(key);
             TState state = storage.GetInteroperable<TState>();
             if (state.Balance < amount) throw new InvalidOperationException();
-            OnBalanceChanging(engine, account, state, -amount);
+            //This method is only called in GAS, so OnBalanceChanging() does nothing.
+            _ = OnBalanceChanging(engine, account, state, -amount);
```

Comment:
Burn should return ContractTask like Mint and Transfer do, and calls to OnBalanceChanging and PostTransfer should be awaited

Reply:
I think this is unnecessary. Because in this case, it must be completed synchronously.